instrumentation/mongo/src/main/java/brave/mongo/TraceMongoCommandListener.java

```diff
@@ -0,0 +1,163 @@
+      .tag("mongo.database", databaseName)
+      .tag("mongo.operation", commandName);
+
+    getAbbreviatedCommand(command).ifPresent(abbreviatedCommand -> span.tag("mongo.command", abbreviatedCommand));
+
+    collectionName.ifPresent(collection -> span.tag("mongo.collection", collection));
+
+    final InetSocketAddress socketAddress = event.getConnectionDescription().getServerAddress().getSocketAddress();
+    span.remoteIpAndPort(socketAddress.getAddress().getHostAddress(), socketAddress.getPort());
+
+    span.start();
+  }
+
+  @Override public void commandSucceeded(final CommandSucceededEvent event) {
+    final Span span = activeSpans.remove(event.getRequestId());
+    if (span == null) {
+      return;
+    }
+    span.finish();
+  }
+
+  @Override public void commandFailed(final CommandFailedEvent event) {
+    final Span span = activeSpans.remove(event.getRequestId());
+    if (span == null) {
+      return;
+    }
+    span.error(event.getThrowable());
+    span.finish();
+  }
+
+  static Optional<String> getCollectionName(final BsonDocument command) {
+    final Iterator<Map.Entry<String, BsonValue>> iterator = command.entrySet().iterator();
+    if (!iterator.hasNext()) {
+      return Optional.empty();
+    }
+    final BsonValue value = iterator.next().getValue();
+    if (!value.isString()) {
+      return Optional.empty();
+    }
+    final String stringValue = value.asString().getValue().trim();
+    return stringValue.isEmpty() ? Optional.empty() : Optional.of(stringValue);
+  }
+
+  /**
+   * Returns an abbreviated version of the command for logging/tracing purposes. Currently this is simply a
+   * truncated version of the command's JSON string representation.
+   *
+   * Note that sensitive data related to the Mongo protocol itself is already scrubbed at this point according to
+   * https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst#security
+   *
+   * @return an abbreviated version of the command for logging/tracing purposes
+   */
+  Optional<String> getAbbreviatedCommand(final BsonDocument command) {
+    if (maxAbbreviatedCommandLength <= 0) {
+      return Optional.empty();
+    }
+    final String abbreviatedCommand = command.toJson(jsonWriterSettings);
+    return abbreviatedCommand.isEmpty() ? Optional.empty() : Optional.of(abbreviatedCommand);
+  }
+
+  static String getSpanName(final String commandName, final Optional<String> collectionName) {
+    return collectionName.map(collection -> commandName + " " + collection).orElse(commandName);
+  }
+
+  public static class Builder {
+    private Tracer tracer;
+    private int maxAbbreviatedCommandLength = 100;
+
+    /**
+     * Sets the {@link Tracer} to use. Required.
+     */
+    public Builder tracer(final Tracer tracer) {
+      this.tracer = tracer;
+      return this;
+    }
+
+    /**
+     * Sets how many characters of the Mongo command to report in the "mongo.command" tag. Defaults to 100.
```

Comment:
this is unusual as we don't do this in any other of our features.. maybe a slight optimization to do that here and now in the first version. I understand that the mongodb.command would be long but so would the sql command, and someone can change data policy on all tags more flexibly with a finishedspanhandler..

Reply:
Not sure but mongo may benefit from this default since SQL in the vast majority of usage will log prepared statements like (INSERT blah VALUES (?)) so while they can get gnarly, it's still sort of a fixed size. If the BsonDocument contains the entire payload of write operations, it's much easier to become ridiculously sized.

Reply:
That being said, 100 seems too small as a default.

Reply:
I tend to agree with @anuraaga here that Mongo is different from SQL here because Mongo commands can contain documents which can be very large (megabytes even), so it probably makes sense to set up sane defaults here.

If somebody wants to, they can set the limit to `Integer.MAX_VALUE` in the builder and then do their own processing in a `FinishedSpanHandler`, but I don't think that we have to make this the easy path.

I'll raise the default limit to 1000 characters.

@adriancole are you okay with keeping the truncation by default, or would you still prefer removing the truncation logic from here?

Reply:
I'm ok with it, just maybe say something in the docs like "set to a very large number ex Integer.MAX_INT" to disable truncation